profiling/src/profile/profiled_endpoints.rs

```diff
@@ -0,0 +1,28 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+use std::collections::HashMap;
+
+use serde::Serialize;
+
+#[derive(Default, PartialEq, Eq, Debug, Clone, Serialize)]
+#[serde(transparent)]
+pub struct ProfiledEndpointsStats {
+    count: HashMap<String, i64>,
+}
+
+impl From<HashMap<String, i64>> for ProfiledEndpointsStats {
+    fn from(count: HashMap<String, i64>) -> Self {
+        ProfiledEndpointsStats { count }
+    }
+}
+
+impl ProfiledEndpointsStats {
+    pub fn add_endpoint_count(&mut self, endpoint_name: String, value: i64) {
+        *self.count.entry(endpoint_name).or_insert(0) += value;
```

Comment:
Btw, not a something to fix but more a general reflexion.

The HashMap::entry API needs to take ownership of the key even if it already exists in the HashMap, which requires allocating a String everytime this function is called, but it's possible to do this with copying only if the key doesn't exists using the [raw_entry api](https://doc.rust-lang.org/std/collections/hash_map/enum.RawEntryMut.html#) (which is currently unstable, but available on hashbrown::HashMap which the std uses underneath) allowing to propagate the Cow<str> to here.
So if allocation or heap fragmentation becomes an issue it might be a good idea to consider this change.

Reply:
Thanks for this.